src/ml_board/frontend/Dashboard/src/worker_socket/WorkerSocket.ts

```diff
@@ -0,0 +1,107 @@
+import socketIO, { Socket } from 'socket.io-client';
+import { settingConfigsInterface } from '../app/App';
+import { DataFromSocket } from './DataTypes';
+import { connectionMTCb, msgCounterIncMTCb, pingMTCb, throughputMTCb, updateMTCb } from './MainThreadCallbacks';
+
+
+// ========================= variables ============================//
+
+// const DEFAULT_URL = 'http://localhost:7000/'; // or http://127.0.0.1:7000/'
+
+let interval: NodeJS.Timer;
+let lastPing: number = -1;
+let lastPong: number = -1;
+const period: number = 10;
+let msgCountPerPeriod: number = 0;
+let socket: Socket;
+
+
+// =~=~=~=~=~=~=~=~=~=~=~=~=~= ~WebSocket~ =~=~=~=~=~=~=~=~=~=~=~=~=~=~=~=//
+const initSocket = (settingConfigs: settingConfigsInterface) => {
+    socket = socketIO(settingConfigs.socketConnectionUrl, { autoConnect: true });
+    console.log("WebSocket initialized");
+    socket.on('connect', () => connect(socket, settingConfigs.gridSearchId));
+    socket.on('disconnect', disconnect);
+    socket.on('connect_error', connect_error);
+    socket.on('mlgym_event', mlgym_event);
+    socket.on('pong', pong);
+};
+
+
+// ========================= connection events ============================//
+const connect = (socket: Socket, runId: string) => {
+    // TODO:ASK how exactly should the join happen? and this was in the old code const runId = "mlgym_event_subscribers";
+    // Max added bug report here: https://github.com/mlgym/mlgym/issues/134
+    socket.emit('join', { rooms: [runId] });
+    // start periodic server pining
+    interval = setInterval(pinging, period * 1000, socket);
+    // flag main thread that connection is on
+    connectionMTCb(true);
+};
+
+const disconnect = (reason: Socket.DisconnectReason) => stop(reason);
+
+const connect_error = (err: Error) => stop(err);
+
+
+// ========================= data driven events ============================//
+// const mlgym_event = (msg:JSON) => {
+const mlgym_event = (msg: string) => {
+    const parsedMsg: DataFromSocket = JSON.parse(msg);
+    // update the redux state on the main thread
+    updateMTCb(parsedMsg);
+    // message count for calculating the throughput
+    msgCountPerPeriod++;
+    // flag main thread to increment the number of incoming messages
+    msgCounterIncMTCb();
+};
+
+const pong = () => {
+    // on Pong , save time of receiving 
+    lastPong = new Date().getTime();
+    // calculate the ping and send it to the MainThread
+    pingMTCb(lastPong - lastPing);
+};
+
+// ========================= helper methods ============================//
+
+const pinging = (socket: Socket) => {
```

Comment:
methods should always indicate an action. 
better: send_ping_to_websocket_server(socket: Socket), which makes it completely clear what the method does. 
There are other methods which sound like variables and should rather indicate an action. Could you check that please?

Reply:
Done!
but I honestly think that this is overly explaining, ping is a clear term on it's own!
I tried to indicate that it's an action with ping_**ing**_ but maybe ping_aciton or ping_process would have been clearer!


I changed pong too, but to make sure I get it right, should I change the other methods too?
e.g.:  connect_error=> onConnect_error, stop=> stopCallback, u.s.w.

Reply:
yeah events should be always follow the pattern `on<Event>`... and these `on<Event>` methods call the respective callback "eventCallback"